src/algebra/big_operators/intervals.lean

```diff
@@ -164,4 +166,119 @@ by rw [← sum_range_id_mul_two n, nat.mul_div_cancel]; exact dec_trivial
 
 end gauss_sum
 
+end generic
+
+section nat
+
+variable {β : Type*}
+variables (f g : ℕ → β) {m n : ℕ}
+
+-- The partial sums (starting from `0`) of `f` and `g` respectively.
+local notation `F` n:80 := ∑ i in range n, f i
+local notation `G` n:80 := ∑ i in range n, g i
+
+section monoid
+variable [add_comm_monoid β]
+
+lemma sum_Ico_empty : (∑ i in Ico n n, f i) = 0 :=
+begin
+  convert sum_empty,
+  exact Ico_self n,
+end
+
+lemma sum_Ico_single : (∑ i in Ico n (n+1), f i) = f n :=
+by rw [sum_eq_sum_Ico_succ_bot (lt_add_one n), sum_Ico_empty, add_zero]
+
+lemma sum_Ico_sub {c: ℕ} (hm : c ≤ m) (hmn : m ≤ n) :
+  ∑ (i : ℕ) in Ico (m-c) (n-c), f (i+c) = ∑ (i : ℕ) in Ico m n, f i :=
```

Comment:
Why is it desirable to put `c` on the right in the index but on the left in the summand? I assumed this was just an oversight from the author of `sum_Ico_add` (perhaps their motivating use-case happened to be in this form?). Is there some logic behind this?

Reply:
@b-mehta. There are now two versions of `sum_Ico_add`, one with the current ordering of `c` (to avoid breakage) and one where `c` appears consistently on the right. I think these lemmas are poorly named (this is not the most important result involving summation and addition), but it's consistent I guess.

I removed `sum_Ico_sub` for now, since there seems to be a strong preference for minimal lines of code/supporting lemmas among Mathlib reviewers. I'm not sure that's the correct choice, since it's more difficult to do rewrites and library search. However, my primary goal is to get this merged so I can start upstreaming the stuff that depends on it.